lib/pure/endians.nim

```diff
@@ -97,17 +121,35 @@ else:
     o[0] = i[1]
     o[1] = i[0]
 
-when system.cpuEndian == bigEndian:
-  proc littleEndian64*(outp, inp: pointer) {.inline.} = swapEndian64(outp, inp)
-  proc littleEndian32*(outp, inp: pointer) {.inline.} = swapEndian32(outp, inp)
-  proc littleEndian16*(outp, inp: pointer) {.inline.} = swapEndian16(outp, inp)
-  proc bigEndian64*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 8)
-  proc bigEndian32*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 4)
-  proc bigEndian16*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 2)
-else:
-  proc littleEndian64*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 8)
-  proc littleEndian32*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 4)
-  proc littleEndian16*(outp, inp: pointer){.inline.} = copyMem(outp, inp, 2)
-  proc bigEndian64*(outp, inp: pointer) {.inline.} = swapEndian64(outp, inp)
-  proc bigEndian32*(outp, inp: pointer) {.inline.} = swapEndian32(outp, inp)
-  proc bigEndian16*(outp, inp: pointer) {.inline.} = swapEndian16(outp, inp)
+when useBuiltinSwap:
+  when system.cpuEndian == bigEndian:
+    proc littleEndian64*(outp, inp: pointer) {.inline.} = swapEndian64(outp, inp)
+    proc littleEndian32*(outp, inp: pointer) {.inline.} = swapEndian32(outp, inp)
+    proc littleEndian16*(outp, inp: pointer) {.inline.} = swapEndian16(outp, inp)
+    proc bigEndian64*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 8)
+    proc bigEndian32*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 4)
+    proc bigEndian16*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 2)
+  else:
+    proc littleEndian64*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 8)
+    proc littleEndian32*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 4)
+    proc littleEndian16*(outp, inp: pointer){.inline.} = copyMem(outp, inp, 2)
+    proc bigEndian64*(outp, inp: pointer) {.inline.} = swapEndian64(outp, inp)
+    proc bigEndian32*(outp, inp: pointer) {.inline.} = swapEndian32(outp, inp)
+    proc bigEndian16*(outp, inp: pointer) {.inline.} = swapEndian16(outp, inp)
+elif defined(js):
```

Comment:
can you remove this entire block by making `copyMem(outp, inp, k)` work for js, so that there's no need to have `elif defined(js):` ?